### PR TITLE
[SYSTEMDS-3801] missing method implementations in ColGroupSDCZeros

### PR DESCRIPTION
**Pull request by e-strauss**
- current master version breaks AWARE reproducibility experiment for kmeans+ algorithm
- missing methods implementations for DenseBlocks in ColGroupSDCZeros
- decreases runtime for kmeans+ algorithm on USCensus dataset from 40s to 32s

### CODE REVIEW

File: src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java [hunk 1]
```diff
@@ -848,23 +848,52 @@ else if(points[c].o < of)
...
 
-		c = processRow(points, sr, nCol, c, of, _data.getIndex(it.getDataIndex()));
+		c = processRowSparse(points, sr, nCol, c, of, _data.getIndex(it.getDataIndex()));
 
 	}
 
 	@Override
 	protected void denseSelection(MatrixBlock selection, P[] points, MatrixBlock ret, int rl, int ru) {
-		throw new NotImplementedException();
+				final DenseBlock dr = ret.getDenseBlock();
```
Comment (Baunsgaard) [inline src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java:857]: Indentation

File: src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java [hunk 2]
```diff
@@ -848,23 +848,52 @@ else if(points[c].o < of)
...
+	private int processRowDense(P[] points, final DenseBlock dr, final int nCol, int c, int of, final int did) {
 		while(c < points.length && points[c].o == of) {
-			_dict.put(sr, did, points[c].r, nCol, _colIndexes);
+			_dict.putDense(dr, did, points[c].r, nCol, _colIndexes);
 			c++;
 		}
 		return c;
 	}
 
+
```
Comment (Baunsgaard) [inline src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java:896]: formatting (no double new line)

**Review by Baunsgaard — COMMENTED** [review]
LGTM, Thanks for the update!
 
There are minor formatting issues that should be easy to resolve, after that, feel free to merge (with an appropriate commit message).

File: src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java (hunk 1 above)
Comment (e-strauss) [reply src/main/java/org/apache/sysds/runtime/compress/colgroup/ColGroupSDCZeros.java:857]: oh yeah, comes from the + signs, the first block I edited in the IDE and + sign confused the IDE to make these wild indentations. After that I edited the block in a separate editor first^^

**Pull request closed without merging**